x/btcstaking/types/parsed_message_validator.go

```diff
@@ -6,29 +6,14 @@ import (
 	"github.com/babylonlabs-io/babylon/btcstaking"
 	bbn "github.com/babylonlabs-io/babylon/types"
 	btcckpttypes "github.com/babylonlabs-io/babylon/x/btccheckpoint/types"
-	"github.com/btcsuite/btcd/btcutil"
 	"github.com/btcsuite/btcd/chaincfg"
-	"github.com/btcsuite/btcd/wire"
 )
 
 type ParamsValidationResult struct {
 	StakingOutputIdx   uint32
 	UnbondingOutputIdx uint32
 }
 
-// caluculateMinimumUnbondingValue calculates minimum unbonding value basend on current staking output value
-// and params.MinUnbondingRate
-func caluculateMinimumUnbondingValue(
-	stakingOutput *wire.TxOut,
-	params *Params,
-) btcutil.Amount {
-	// this conversions must always succeed, as it is part of our params
-	minUnbondingRate := params.MinUnbondingRate.MustFloat64()
-	// Caluclate min unbonding output value based on staking output, use btc native multiplication
-	minUnbondingOutputValue := btcutil.Amount(stakingOutput.Value).MulF64(minUnbondingRate)
-	return minUnbondingOutputValue
-}
-
 // ValidateParams validates parsed message against parameters
 func ValidateParams(
```

Comment:
agree naming is a bit confusing.

Though I would like to avoid `ParsedCreateDelegationMessage` to have any funcion on it, as in my mind it should remain as simple object that contains just data.

Maybe updating name to `ValidateParsedMessageAgainstTheParams`  would be less confusing then ?